### ISSUE TITLE
Support Projects feature

### ISSUE THREAD

**Issue by rupaschomaker** [devopshq/artifactory#257]
Please consider adding support for Projects:

https://www.jfrog.com/confluence/display/JFROG/Artifactory+REST+API#ArtifactoryRESTAPI-PROJECTS

Note, must use Bearer Token when using this api.  Also, requires /access endpoint, not /artifactory and things like creating a repo in a particular project requires a key "projectKey".

**Comment by zhan9san** [comment]
Let me try to add this feature.

**Comment by allburov** [comment]
Is it "Admin area" objects? If it's - let's add it here https://github.com/devopshq/artifactory/blob/master/dohq_artifactory/admin.py

@zhan9san It'd be great!

**Comment by rupaschomaker** [comment]
I was wrong, the /api/repositories and /api/repositores/{key} endpoints do not have any knowledge of projectKey -- I have an open ticket with jfrog on that.

The functionality to be added all make sense in admin.py.

**Comment by zhan9san** [comment]
> Is a project "Admin area" object? If it's - let's add it here https://github.com/devopshq/artifactory/blob/master/dohq_artifactory/admin.py
> 
> @zhan9san It'd be great!

Yes. It's an "Admin area" object. I have implemented CURD of Projects, and added token support in `dohq_artifactory/auth.py`.

Projects(Done)

1. Get Projects List
2. Add a New Project
3. Get Project
4. Update Existing Project Properties
5. Delete Existing Project

---

Project Users(In-progress)

1. Get Project Users
2. Get Project User
3. Update User in Project
4. Delete Existing Project's User

---

Project Groups

1. Get Projects Groups
2. Get Projects Group
3. Update Group in Project
4. Delete Existing Project's Group

---

Project Roles

1. Get Roles List
2. Add a New Role
3. Get Project Role
4. Update an Existing Project Role
5. Delete an Existing Role

---

Projecct Repository

1. Move Repository in a Project
2. Share Repository with All Projects
3. Unshare Repository with All Projects
4. Share Repository with Target Project
5. Unshare Repository with Target Project
6. Unassign a Project from a Repository

I'll add

**Comment by beliaev-maksim** [comment]
@zhan9san 
is the work on this item completed? can we close it?

**Comment by zhan9san** [comment]
The basic functionality has been implemented in #266 .

If someone wants the rest, let us trace it in another small issue.

Feel free to close this issue.
